core/zip/inc/Compression.h

```diff
@@ -31,6 +31,15 @@ namespace ROOT {
 /// The current algorithms support level 1 to 9. The higher the level the greater
 /// the compression and more CPU time and memory resources used during compression.
 /// Level 0 means no compression.
+///
+/// Recommendation for the compression algorithm's levels:
+///  - ZLIB is recommended to be used with compression level 6 [106] 
+///  - LZMA is recommended to be used with compression level 7-8 (higher is better, 
+///   since in the case of LZMA we don't care about compression/decompression speed)
+///   [207 - 208]
+///  - LZ4 is recommended to be used with compression level 4 [404]
+
```

Comment:
It might be nice to add a
```
constexpr const int kDefaultCompressionLevel[kUndefinedCompressionAlgorithm] = {-1,  6, 7, ?, 4};
```
and use that in the code (instead of the current default values that I suppose we have, of some sort, in one or more places?)

@pcanal @bbockelm what's your take?

Reply:
I think that's a great way to go!

To be honest, I'm not even sure of all the different places these "magic integers" can be found -- discovering and centralizing would be very useful.

That said, the other thing we need to do is to minimize use of our bizarre `106` syntax which makes the compression subsystem inaccessible to everyone but experts.

Reply:
Yes, I agree :)